src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/jpa/repository/NonAssociationsRepository.kt

```diff
@@ -6,7 +6,12 @@ import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation
 
 @Repository
 interface NonAssociationsRepository : JpaRepository<NonAssociation, Long> {
-  fun findAllByFirstPrisonerNumber(prisonerNumber: String): List<NonAssociation>
-  fun findAllBySecondPrisonerNumber(prisonerNumber: String): List<NonAssociation>
+  fun findAllByFirstPrisonerNumberOrSecondPrisonerNumber(firstPrisonerNumber: String, secondPrisonerNumber: String): List<NonAssociation>
   fun findByFirstPrisonerNumberAndSecondPrisonerNumber(firstPrisonerNumber: String, secondPrisonerNumber: String): NonAssociation?
 }
+
+/**
+ * Where the given prisoner number is either first OR second in the non-association
+ */
+fun NonAssociationsRepository.findByPrisonerNumber(prisonerNumber: String): List<NonAssociation> =
+  findAllByFirstPrisonerNumberOrSecondPrisonerNumber(prisonerNumber, prisonerNumber)
```

Comment:
fyi: needs to be an extension function rather than a concrete method inside the interface because spring still tries to turn it into a materialised method otherwise